python/packages/autogen-ext/src/autogen_ext/tools/graphrag/_model_adapter.py

```diff
@@ -1,40 +1,47 @@
+# mypy: disable-error-code="no-any-unimported,misc"
 import asyncio
 import json
-from typing import Any, List, Optional, Union
+from typing import Any, AsyncGenerator, Dict, Generator, List, Union, cast
 
-from graphrag.query.llm.base import BaseLLM
-
-from autogen_ext.models.openai._openai_client import (
+from autogen_core.models import (
     AssistantMessage,
     CreateResult,
-    OpenAIChatCompletionClient,
     SystemMessage,
     UserMessage,
 )
+from autogen_ext.models.openai import OpenAIChatCompletionClient
+from autogen_ext.models.openai._openai_client import AzureOpenAIChatCompletionClient
+
+from graphrag.query.llm.base import BaseLLM, BaseLLMCallback
 
 
 class GraphragOpenAiModelAdapter(BaseLLM):
     """
     Adapts an autogen OpenAIChatCompletionClient to a graphrag-compatible LLM interface.
     """
 
-    def __init__(self, client: OpenAIChatCompletionClient):
+    def __init__(self, client: OpenAIChatCompletionClient | AzureOpenAIChatCompletionClient):
```

Comment:
Since we have already adopted so much of GraphRAG's components here, do you think it makes sense to just use the BaseLLM type instead of the `ChatCompletionClient` type and creating an adaptor?

Reply:
One of the things I tried to do is add a factory method where the user doesn't need to be exposed to graphrag components with `from_config`. This is what I'm using in the example script. The constructor right now accepts the graphrag components to allow more flexibility and to allow users to create the tool even if we forgot to add support to some customisation option in the factory. Still needs to be improved, but that is the goal. In the script I can create the tool just from our llm client + configuration options.

Reply:
I think it makes sense though I am hoping to avoid too much code bloat. Checking with @jackgerrits

Reply:
Alright, I replaced it with a `from_settings` method that uses the graphrag configuration file. Should be much clear now.